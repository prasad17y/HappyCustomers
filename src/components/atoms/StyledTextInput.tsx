import React from 'react';
import {TextInput, StyleSheet, View, TextInputProps} from 'react-native';

interface StyledTextInputProps extends TextInputProps {
  hasError?: boolean;
}

const StyledTextInput: React.FC<StyledTextInputProps> = ({
  hasError,
  style,
  ...props
}) => {
  return (
    <View style={[styles.inputContainer, hasError && styles.inputError]}>
      <TextInput
        style={[styles.input, style]}
        placeholderTextColor="#a0a0a0"
        {...props}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  inputContainer: {
    borderBottomWidth: 1,
    borderBottomColor: '#e0e0e0',
  },
  input: {
    fontSize: 16,
    paddingVertical: 12,
    color: '#191a1a',
  },
  inputError: {
    borderBottomColor: 'red',
  },
});

export default React.memo(StyledTextInput);
