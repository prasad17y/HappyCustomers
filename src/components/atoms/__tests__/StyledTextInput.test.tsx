import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react-native';
import StyledTextInput from '../StyledTextInput';

describe('StyledTextInput', () => {
  it('should render correctly and match the snapshot', () => {
    const tree = render(
      <StyledTextInput value="Test" placeholder="Enter text" />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should handle text changes', () => {
    const onChangeTextMock = jest.fn();
    render(
      <StyledTextInput
        value=""
        onChangeText={onChangeTextMock}
        placeholder="Enter text"
      />,
    );

    const input = screen.getByPlaceholderText('Enter text');
    fireEvent.changeText(input, 'hello world');
    expect(onChangeTextMock).toHaveBeenCalledWith('hello world');
  });

  it('should apply an error style when hasError is true', () => {
    const tree = render(
      <StyledTextInput value="Test" placeholder="Enter text" hasError />,
    );

    // The snapshot will now include the red border color, locking in the error style
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
