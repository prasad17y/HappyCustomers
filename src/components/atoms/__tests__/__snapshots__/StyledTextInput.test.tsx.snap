// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledTextInput should apply an error style when hasError is true 1`] = `
<View
  style={
    [
      {
        "borderBottomColor": "#e0e0e0",
        "borderBottomWidth": 1,
      },
      {
        "borderBottomColor": "red",
      },
    ]
  }
>
  <TextInput
    placeholder="Enter text"
    placeholderTextColor="#a0a0a0"
    style={
      [
        {
          "color": "#191a1a",
          "fontSize": 16,
          "paddingVertical": 12,
        },
        undefined,
      ]
    }
    value="Test"
  />
</View>
`;

exports[`StyledTextInput should render correctly and match the snapshot 1`] = `
<View
  style={
    [
      {
        "borderBottomColor": "#e0e0e0",
        "borderBottomWidth": 1,
      },
      undefined,
    ]
  }
>
  <TextInput
    placeholder="Enter text"
    placeholderTextColor="#a0a0a0"
    style={
      [
        {
          "color": "#191a1a",
          "fontSize": 16,
          "paddingVertical": 12,
        },
        undefined,
      ]
    }
    value="Test"
  />
</View>
`;
